pkg/monitortests/kubeapiserver/auditloganalyzer/monitortest/clustermonitor.go

```diff
@@ -0,0 +1,127 @@
+	client, err := kubernetes.NewForConfig(m.config)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var errs []error
+
+	// first, process the audit events
+	eventsProvider := eventsprovider.NewClusterEvents(&beginning, &end)
+	collectors := auditloganalyzer.AuditEventCollectors(m.collectors)
+	for e := range eventsProvider.Events(ctx, client) {
+		collectors.Collect(e)
+	}
+	errs = append(errs, eventsProvider.Err())
+
+	// collect points of interest generated from processing the audit log events
+	var rawIntervals []monitorapi.Interval
+	var tests []*junitapi.JUnitTestCase
+	for _, collector := range collectors {
+		if inspector, ok := collector.(auditloganalyzer.TestDataCollector); ok {
+			intervals, units, err := inspector.InspectTestArtifacts(ctx, storageDir, beginning, end)
+			rawIntervals = append(rawIntervals, []monitorapi.Interval(intervals)...)
+			tests = append(tests, units...)
+			errs = append(errs, err)
+		}
+		if inspector, ok := collector.(auditloganalyzer.ClusterDataCollector); ok {
+			intervals, units, err := inspector.InspectCluster(ctx, beginning, end)
+			rawIntervals = append(rawIntervals, []monitorapi.Interval(intervals)...)
+			tests = append(tests, units...)
+			errs = append(errs, err)
+		}
+	}
+	return rawIntervals, tests, errors.Join(errs...)
+}
+
+func (m *clusterMonitor) ConstructComputedIntervals(ctx context.Context, startingIntervals monitorapi.Intervals, recordedResources monitorapi.ResourcesMap, beginning, end time.Time) (monitorapi.Intervals, error) {
+	var computedIntervals []monitorapi.Interval
+	var errs []error
+	for _, collector := range m.collectors {
+		if computer, ok := collector.(auditloganalyzer.IntervalComputer); ok {
+			intervals, err := computer.ProcessAggregatedIntervals(ctx, startingIntervals, recordedResources, beginning, end)
+			computedIntervals = append(computedIntervals, []monitorapi.Interval(intervals)...)
+			errs = append(errs, err)
+		}
+	}
+	return computedIntervals, errors.Join(errs...)
+}
+
+func (m *clusterMonitor) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	var tests []*junitapi.JUnitTestCase
+	var errs []error
+	for _, collector := range m.collectors {
+		if computer, ok := collector.(auditloganalyzer.TestEvaluator); ok {
+			units, err := computer.Evaluate(ctx, finalIntervals)
+			tests = append(tests, units...)
+			errs = append(errs, err)
+		}
+	}
+	return tests, errors.Join(errs...)
+}
+
+func (m *clusterMonitor) WriteContentToStorage(ctx context.Context, storageDir, timeSuffix string, finalIntervals monitorapi.Intervals, finalResourceState monitorapi.ResourcesMap) error {
+	var errs []error
+	for _, collector := range m.collectors {
+		if computer, ok := collector.(auditloganalyzer.StorageContentWriter); ok {
+			err := computer.SaveArtifacts(ctx, storageDir, timeSuffix, finalIntervals, finalResourceState)
+			errs = append(errs, err)
+		}
+	}
```

Comment:
same here.

I'll say if we break six things that need this, then we should make an interface.